tools/scxtop/src/config.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+//
+// This software may be used and distributed according to the terms of the
+// GNU General Public License version 2.
+
+use crate::cli::Cli;
+use crate::STATS_SOCKET_PATH;
+use crate::TRACE_FILE_PREFIX;
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::path::{Path, PathBuf};
+use xdg;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Config {
+    /// App tick rate in milliseconds.
+    pub tick_rate_ms: usize,
+    /// Extra verbose output.
+    pub debug: bool,
+    /// Exclude bpf event tracking.
+    pub exclude_bpf: bool,
+    /// Stats unix socket path.
+    pub stats_socket_path: String,
+    /// Trace file prefix for perfetto traces.
+    pub trace_file_prefix: String,
+    /// Number of ticks for traces.
+    pub trace_ticks: usize,
+    /// Number of worker threads
+    pub worker_threads: u16,
+    /// Number of ticks to warmup before collecting traces.
+    pub trace_tick_warmup: usize,
+}
+
+pub fn get_config_path() -> Result<PathBuf> {
+    let xdg_dirs = xdg::BaseDirectories::with_prefix("scxtop")?;
+    let config_path = xdg_dirs.get_config_file("scxtop.toml");
+    Ok(config_path)
+}
+
+impl Config {
+    /// Returns the default config.
+    pub fn default_config() -> Config {
+        Config {
+            tick_rate_ms: 250,
+            debug: false,
+            exclude_bpf: false,
+            stats_socket_path: STATS_SOCKET_PATH.to_string(),
+            trace_file_prefix: TRACE_FILE_PREFIX.to_string(),
+            trace_ticks: 5,
+            worker_threads: 4,
+            trace_tick_warmup: 3,
+        }
+    }
```

Comment:
Could we add these defaults using the `serde::default` tags like in layered? Then if the config isn't present we could parse an empty string and get the default config, else fields would be filled in with defaults if they aren't specified. What do you think? I prefer the idea of being able to encode only what I want to specify in the config file and leaving the rest to the app defaults that may change over time.

Reply:
That sounds good!